### PR TITLE
Add ChatMessage and ChatHistory util classes

### PR DESCRIPTION
**Pull request by evangriffiths**

### CODE REVIEW

File: prediction_market_agent/agents/microchain_agent/memory.py [hunk 1]
```diff
@@ -99,3 +95,78 @@ def search(
...
+class ChatHistory(BaseModel):
+    """
+    A collection of chat messages. Assumes that the chat messages are ordered by
+    datetime.
+    """
+
+    chat_messages: Sequence[ChatMessage]
+
+    def add_message(self, chat_message: ChatMessage) -> None:
+        list(self.chat_messages).append(chat_message)
```
[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent/agents/microchain_agent/memory.py:109]: Rabbit is correct, this won't modify `chat_messages` of this model. 

Why is chat_messages typed as Sequence, are there some errors if it's a list? (sequence is unmutable)

File: prediction_market_agent/agents/microchain_agent/memory.py [hunk 2]
```diff
@@ -99,3 +95,78 @@ def search(
...
+
+    @staticmethod
+    def from_list_of_dicts(list_of_dicts: list[Dict[str, str]]) -> "ChatHistory":
+        chat_messages = [
+            ChatMessage(role=str(h["role"]), content=str(h["value"]))
```
Comment (kongzii) [inline prediction_market_agent/agents/microchain_agent/memory.py:114]: list_of_dicts is typed as `list_of_dicts: list[Dict[str, str]]`, so these extra `str(...)` calls should be unnecessary, unless there is something hidden I don't see

File: prediction_market_agent/agents/microchain_agent/memory.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent/agents/microchain_agent/memory.py:109]: that was fixed in an earlier commit (https://github.com/gnosis/prediction-market-agent/pull/260/commits/9f37a5785cefbf321b7f8cbb28b178a5e0f0336a) 😃

[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent/agents/microchain_agent/memory.py:109]: Is it? I can still see it in the latest changes that would go to main:

Screenshot by Dropbox Capture

File: prediction_market_agent/agents/microchain_agent/memory.py (hunk 2 above)
Comment (evangriffiths) [reply prediction_market_agent/agents/microchain_agent/memory.py]: You're right, removed.

File: prediction_market_agent/agents/microchain_agent/memory.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent/agents/microchain_agent/memory.py]: Ah sorry. Use of function was removed but left this in by accident. Thanks!

**Review by kongzii — APPROVED** [review]

**Pull request merged**